topostats/__init__.py

```diff
@@ -22,3 +29,244 @@
 
 # Disable snoop
 snoop.install(enabled=False)
+
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-positional-arguments
+
+
+@dataclass
+class TopoStats:
+    """
+    Class for storing TopoStats objects.
+
+    Attributes
+    ----------
+    image_grain_crops : ImageGrainCrops | None
+        ImageGrainCrops of processed image.
+    filename : str | None
+        Filename.
+    pixel_to_nm_scaling : str | None
+        Pixel to nanometre scaling.
+    img_path : str | None
+        Original path to image.
+    image : npt.NDArray | None
+        Flattened image (post ''Filter()'').
+    image_original : npt.NDArray | None
+        Original image.
+    topostats_version : str | None
+        TopoStats version.
+    """
+
+    image_grain_crops: ImageGrainCrops | None
+    filename: str | None
+    pixel_to_nm_scaling: str | None
+    img_path: Path | str | None
+    image: npt.NDArray | None
+    image_original: npt.NDArray | None
+    topostats_version: str | None
+
+    def __eq__(self, other: object) -> bool:
+        """
+        Check if two TopoStats objects are equal.
+
+        Parameters
+        ----------
+        other : object
+          Object to compare to.
+
+        Returns
+        -------
+        bool
+          True if the objects are equal, False otherwise.
+        """
+        if not isinstance(other, TopoStats):
+            return False
+        return (
+            self.image_grain_crops == other.image_grain_crops
+            and self.filename == other.filename
+            and self.pixel_to_nm_scaling == other.pixel_to_nm_scaling
+            and self.topostats_version == other.topostats_version
+            and self.img_path == other.image_path
+            and self.image == other.image
+            and self.image_original == other.image_original
```

Comment:
Hm.. This seems like it shouldn't work..? 

Testing equality between two `npt.NDArray` objects is not allowed via `==` I thought
 
<img width="733" alt="image" src="https://github.com/user-attachments/assets/cbeec17a-4e25-41d8-83c7-c077d010bee3" />

Reply:
Very good point, tests duly written and we're now using `np.all(self.image == other.image)` to check for equality.